pkg/cloud/aws/actuators/machine/actuator.go

```diff
@@ -158,26 +117,82 @@ func (a *Actuator) Create(ctx context.Context, cluster *clusterv1.Cluster, machi
 	}
 
 	controlPlaneMachines := GetControlPlaneMachines(clusterMachines)
+	if len(controlPlaneMachines) == 0 {
+		log.Info("No control plane machines exist yet - requeuing")
+		return &controllerError.RequeueAfterError{RequeueAfter: 5 * time.Second}
+	}
+
+	// Except for the very first control plane machine, everything will be a join, so let's assume we're joining.
+	isNodeJoin := true
+	if cluster.Annotations[v1alpha1.AnnotationControlPlaneReady] != "true" {
+		if machine.Spec.Versions.ControlPlane == "" {
+			// This isn't a control plane machine - have to wait
+			log.Info("No control plane machines exist yet - requeuing")
+			return &controllerError.RequeueAfterError{RequeueAfter: 5 * time.Second}
+		}
 
-	isNodeJoin, err := a.isNodeJoin(scope, controlPlaneMachines)
-	if err != nil {
-		return errors.Wrapf(err, "failed to determine whether machine %q should join cluster %q", machine.Name, cluster.Name)
+		configMapName := actuators.ControlPlaneConfigMapName(cluster)
+		log.Info("Checking for existence of control plane configmap lock", "configmap-name", configMapName)
+
+		controlPlaneConfigMap, err := a.coreClient.ConfigMaps(cluster.Namespace).Get(configMapName, metav1.GetOptions{})
+		if apierrors.IsNotFound(err) {
+			controlPlaneConfigMap = &apicorev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: cluster.Namespace,
+					Name:      configMapName,
+					OwnerReferences: []metav1.OwnerReference{
+						{
+							APIVersion: cluster.APIVersion,
+							Kind:       cluster.Kind,
+							Name:       cluster.Name,
+							UID:        cluster.UID,
+						},
+					},
+				},
+			}
+
+			log.Info("Attempting to create control plane configmap lock")
+			_, err = a.coreClient.ConfigMaps(cluster.Namespace).Create(controlPlaneConfigMap)
+			if err != nil {
+				if apierrors.IsAlreadyExists(err) {
+					// Someone else beat us to it
+					log.Info("Control plane configmap lock already exists - requeuing")
+				} else {
+					log.Error(err, "Error creating control plane configmap lock")
+				}
+
+				return &controllerError.RequeueAfterError{RequeueAfter: 5 * time.Second}
+			}
+
+			// This is the first control plane machine, so it's not a join.
+			isNodeJoin = false
+		} else if err != nil {
+			return errors.Wrap(err, "error getting control plane configmap lock")
+		} else {
+			// err is nil, which means the configmap exists but the control plane isn't ready yet.
+			// Need to requeue.
+			log.Info("Control plane configmap lock already exists but control plane is not ready yet - requeuing")
+			return &controllerError.RequeueAfterError{RequeueAfter: 5 * time.Second}
+		}
 	}
 
 	var bootstrapToken string
 	if isNodeJoin {
-		a.log.V(2).Info("Machine will join the cluster", "cluster", cluster.Name, "machine-name", machine.Name, "machine-namespace", machine.Namespace)
+		// First control plane machine has already been created. Now we can't proceed until its apiserver is up.
```

Comment:
This comment doesn't seem accurate, since the waiting is done prior to this point.